src/util/base64.cpp

```diff
@@ -33,16 +33,34 @@ static const std::string base64_chars =
 		"abcdefghijklmnopqrstuvwxyz"
 		"0123456789+/";
 
+static const std::string base64_chars_padding_1 = "AEIMQUYcgkosw048";
+static const std::string base64_chars_padding_2 = "AQgw";
 
 static inline bool is_base64(unsigned char c)
 {
-	return isalnum(c) || c == '+' || c == '/' || c == '=';
+	return (c >= '0' && c <= '9')
+			|| (c >= 'A' && c <= 'Z')
+			|| (c >= 'a' && c <= 'z')
+			|| c == '+' || c == '/';
 }
 
 bool base64_is_valid(std::string const& s)
 {
-	for (char i : s)
-		if (!is_base64(i))
+	int i = 0;
+	for (; i < s.size(); ++i)
+		if (!is_base64(s[i]))
+			break;
+	// number of padding characters needs to match
+	unsigned char padding = 3 - ((s.size() + 3) % 4);
+	int actual_padding = s.size() - i;
+	if ((actual_padding != padding && actual_padding != 0)
```

Comment:
shouldn't `actual_padding == 0` cause none of these checks to be ran?

Reply:
well, it doesn't matter. I could return early if that's what you want.

Reply:
Maybe it already works correctly but I had trouble understanding the logic to verify its' correctness